smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java

```diff
@@ -110,9 +110,11 @@ void afterDeploymentValidation(@Observes AfterDeploymentValidation done, BeanMan
             for (InjectionPoint ip : emitterInjectionPoints) {
                 String name = ChannelProducer.getChannelName(ip);
                 EmitterImpl<?> emitter = (EmitterImpl<?>) registry.getEmitter(name);
-                if (!emitter.isConnected()) {
-                    done.addDeploymentProblem(
-                            new DeploymentException("No channel found for name: " + name + ", injection point: " + ip));
+                if (!emitter.isSubscribed()) {
+                    // Subscription may happen later, just print a warning.
+                    // Attempting an emission without being subscribed would result in an error.
+                    LOGGER.warn("No subscriber for channel {}  attached to the emitter {}.{}", name,
```

Comment:
Warning or just a trace?

How important is it for developers/debugging to know nothing is subscribed on creation?

Reply:
It can be important. In the previous version, it was failing the _deployment_, that's why I went for warning.